nemo/lightning/data.py

```diff
@@ -209,6 +207,32 @@ def __iter__(self): ...
 
 
 class MegatronPretrainingSampler(BaseMegatronSampler):
+    def __init__(
+        self,
+        total_samples: int,
+        consumed_samples: int,
+        micro_batch_size: int,
+        data_parallel_rank: int,
+        data_parallel_size: int,
+        drop_last: bool = True,
+        global_batch_size: Optional[int] = None,
+        rampup_batch_size: Optional[list] = None,
+        pad_samples_to_global_batch_size: Optional[bool] = False,
```

Comment:
Any reason to list them out seperately instead of just
```python
def __init__(self, *args, **kwargs):
     super().__init__(*args, **kwargs)
```

Reply:
Just makes the args more explicit. I don' have much of a preference; can update it if you prefer

Reply:
No preference, this works for me then.

Reply:
I guess it might be a bit difficult to maintain if more args get added to the underlying Megatron class

Reply:
That's true. Maybe we can address this in a separate PR? The [MegatronPretrainingRandomSampler](https://github.com/NVIDIA/NeMo/blob/main/nemo/lightning/data.py#L234) init is the same, so if we make the change to the `MegatronPretrainingSampler`, we should probably do the same to the random sampler.

Reply:
Cool sounds good.